internalv2compat/compat.go

```diff
@@ -63,3 +64,19 @@ type IsHTTP interface {
 type IsThrift interface {
 	isThrift()
 }
+
+var (
+	v2TracingEnabledMutex sync.RWMutex
+	v2TracingEnabled      bool
+)
+
+func SetV2TracingEnabled(enabled bool) {
+	v2TracingEnabledMutex.Lock()
+	defer v2TracingEnabledMutex.Unlock()
+	v2TracingEnabled = enabled
+}
+func V2TracingEnabled() bool {
+	v2TracingEnabledMutex.RLock()
+	defer v2TracingEnabledMutex.RUnlock()
+	return v2TracingEnabled
+}
```

Comment:
* I don't think you need an RWMutex here -- they are a bit slower, and are only really useful if you _need_ the multiple readers for throughput
* I often like combining the "locked thing" in with the mutex.

Together it could look like

```suggestion
var v2Tracing struct{
  sync.Mutex
	enabled bool
}

func SetV2TracingEnabled(enabled bool) {
	v2Tracing.Lock()
	defer v2Tracing.Unlock()
	v2Tracing.Enabled = enabled
}
func V2TracingEnabled() bool {
	v2Tracing.Lock()
	defer v2Tracing.Unlock()
	return v2Tracing.Enabled
}
```

Reply:
Refactored.